src/content-feature.js

```diff
@@ -94,33 +92,38 @@ export default class ContentFeature {
     }
 
     /**
-     * @returns {MessagingContext}
+     * @deprecated as we should make this internal to the class and not used externally
+     * @return {MessagingContext}
      */
-    get messagingContext () {
-        if (this.#messagingContext) return this.#messagingContext
-
-        const contextName = import.meta.injectName === 'apple-isolated'
+    _createMessagingContext () {
+        const injectName = import.meta.injectName
+        if (typeof injectName === 'undefined') throw new Error('import.meta.injectName missing')
+        const contextName = injectName === 'apple-isolated'
             ? 'contentScopeScriptsIsolated'
             : 'contentScopeScripts'
 
-        this.#messagingContext = new MessagingContext({
+        return new MessagingContext({
             context: contextName,
             env: this.isDebug ? 'development' : 'production',
             featureName: this.name
         })
-        return this.#messagingContext
     }
 
-    // Messaging layer between the content feature and the Platform
-    get debugMessaging () {
-        if (this.#debugMessaging) return this.#debugMessaging
-
-        if (this.platform?.name === 'extension' && typeof import.meta.injectName !== 'undefined') {
-            this.#debugMessaging = createMessaging({ name: 'debug', isDebug: this.isDebug }, import.meta.injectName)
```

Comment:
Note that debug messaging was separated because it had a separate 'feature' name (`'debug'` instead of `this.name`). This was supposed to allow the native platform side write a single message handler for breakage flags instead of duplicating the same one per every C-S-S feature (even those that don't really use messaging otherwise).
The only consumer at the moment is the extension though, where this is not a big deal, but on other platforms it could be important

Reply:
I think we can approach that when we come to it. I don't really know each platforms integrations but that sounds a little fragile if they're implemented that way.

Either way we can create a singleton and pass it in but we'd risk a tonne of duplication again.

Let's get @shakyShane to confirm, I'd rather we don't do this for now.